connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala

```diff
@@ -64,6 +69,94 @@ private[client] object GrpcExceptionConverter extends JsonUtils {
     }
   }
 
+  private def enrichError(errors: Array[Error], info: ErrorInfo): Array[Error] = {
+    val sessionId = info.getMetadataOrDefault("sessionId", null)
+    val userId = info.getMetadataOrDefault("userId", null)
+
+    if (sessionId == null || userId == null) {
+      return errors
+    }
+
+    try {
+      val errorInfoResponse = stub.getErrorInfo(
+        GetErrorInfoRequest
+          .newBuilder()
+          .setSessionId(sessionId)
+          .addAllErrorIds(errors.map(_.errorId).toIterable.asJava)
+          .setUserContext(UserContext.newBuilder().setUserId(userId).build())
+          .build())
+
+      val exceptionInfos = errorInfoResponse.getExceptionInfosList.asScala
+
+      errors.map { error =>
+        exceptionInfos.find(_.getErrorId == error.errorId) match {
+          case Some(exceptionInfo) =>
+            val stackTraceOpt =
+              if (exceptionInfo.getStackTraceList.isEmpty) None
+              else {
+                Some(exceptionInfo.getStackTraceList.asScala.toArray.map { stackTraceElement =>
+                  new StackTraceElement(
+                    stackTraceElement.getDeclaringClass,
+                    stackTraceElement.getMethodName,
+                    stackTraceElement.getFileName,
+                    stackTraceElement.getLineNumber)
+                })
+              }
+            Error(
+              errorId = error.errorId,
+              classes = error.classes,
+              message = exceptionInfo.getMessage,
+              stackTraceOpt = stackTraceOpt)
+          case None =>
+            error
+        }
+      }
+    } catch {
+      case _: StatusRuntimeException => errors
```

Comment:
Do we want to retry here?

Reply:
The current expectation on this additional rpc call is best effort and at most once so no?